### DISCUSSION TITLE
YOLO model for detecting cats, foxes, and dogs in urban gardens

### DISCUSSION THREAD

**Discussion by aardrasystems** [ultralytics/ultralytics#22041] [Q&A]
Hi All,

I am very new to Ultralytics and YOLO. I am aware there are many forks in the Ultralytics GitHub library. Has anyone worked on a model that can detect animals such as cats, foxes, and dogs in urban settings, for example, a garden?

I am looking for a starting point, as the standard model does not always seem to detect every variation.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @aardrasystems, thanks for reaching out and welcome to Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also assist you here soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) for a quick overview, including many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples where common questions are often answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please share:
- Your exact command or code snippet
- Model and version used (e.g., YOLO11n/s/m, pretrained weights)
- Sample images from your target environment (urban gardens) and any labels
- Training/evaluation logs and metrics
- Inference settings (image size, confidence/NMS thresholds)
And verify you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Great question—start with Ultralytics YOLO11 and fine-tune on your own garden images; COCO covers `cat` and `dog`, but `fox` typically needs a few labeled examples from your scene for reliable recall. See the [YOLO11 models overview](https://docs.ultralytics.com/models/yolo11/) and the [detection dataset format guide](https://docs.ultralytics.com/datasets/detect/) for how to structure data; this end‑to‑end walkthrough is also helpful: [training YOLO11 to detect animals](https://www.ultralytics.com/blog/how-to-train-ultralytics-yolo-models-to-detect-animals-in-the-wild).

```yaml
# data.yaml
path: /data/garden-animals
train: images/train
val: images/val
names: [cat, dog, fox]
```

```python
from ultralytics import YOLO
model = YOLO("yolo11s.pt")  # good recall/speed balance
model.train(data="data.yaml", epochs=50, imgsz=640)  # fine-tune on your images
model.predict(source="your_garden_video.mp4", conf=0.2, iou=0.6, imgsz=832)  # test with lower conf/higher imgsz
```

If you can share a couple of representative annotated images (day/night, different distances), we can suggest targeted tweaks; otherwise this workflow is the proven starting point.
